x-pack/plugins/security_solution/server/endpoint/routes/trusted_apps/handlers.ts

```diff
@@ -47,3 +48,32 @@ export const getTrustedAppsListRouteHandler = (
     }
   };
 };
+
+export const getTrustedAppsCreateRouteHandler = (
+  endpointAppContext: EndpointAppContext
+): RequestHandler<undefined, undefined, PostTrustedAppCreateRequest> => {
+  const logger = endpointAppContext.logFactory.get('trusted_apps');
+
+  return async (constext, req, res) => {
+    const exceptionsListService = endpointAppContext.service.getExceptionsList();
+    const newTrustedApp = req.body;
+
+    try {
+      // Ensure list is created if it does not exist
+      await exceptionsListService?.createTrustedAppsList();
```

Comment:
this will check if the list is created first?

Reply:
Yes - the `createTrustedAppsLists()` method will not create it if it already exists - a warning is logged to the kibana log. Ref: https://github.com/elastic/kibana/blob/d46227421e361fd5ea29cbb6e512c9df66448d5e/x-pack/plugins/lists/server/services/exception_lists/create_endpoint_trusted_apps_list.ts#L70-L72

(FYI: I just noticed the `?` in the above code - will remove it (no longer needed)